docs/multicall-notes.md

```diff
@@ -0,0 +1,216 @@
+
+| parameter name | default value |
+-----------------|-----------------------
+| prevRandao | 0x0 |
+| feeRecipient | 0x0 |
+| mixHash | 0x0 |
+| nonce | 0x0 |
+| extraData | 0x0 |
+| difficulty | The same as the base block defined as the second parameter in the call |
+| gasLimit | The same as the base block defined as the second parameter in the call |
+| hash | calculated normally, except for phantom blocks, see below Phantom block section |
+| parentHash | previous blocks hash (the real hash, or phantom blocks hash) |
+| timestamp | the timestamp of previous block + 1 |
+| baseFeePerGas | calculated on what it should be according to ethereum's spec. Note: Phantom blocks do not have transactions so this is always adjusted downwards on them |
+| sha3Uncles | empty trie root |
+| withdrawals | empty array |
+| uncles | empty array |
+| logsBloom | calculated normally. ETH logs are not part of the calculation |
+| receiptsRoot | calculated normally |
+| transactionsRoot | calculated normally |
+| number | calculated normally |
+| size | calculated normally |
+| withdrawalsRoot | calculated normally |
+| gasUsed | calculated normally |
+| stateRoot | calculated normally |
+
+An interesting note here is that we decide timestamp as `previous block timestamp + 1`, while `previous block timestamp + 12` could also be an assumed default. The reasoning to use `+1` is that it's the minimum amount we have to increase the timestamp to keep them valid. While `+12` is what Mainnet uses, there are other chains that use some other values, and we didn't want to complicate the specification to consider all networks.
+
+## Phantom blocks
+The multicall allows you to define on what block number your calls or transactions are being executed on. Eg, consider following call:
+```json
+{
+	"jsonrpc": "2.0",
+	"id": 1,
+	"method": "eth_multicallV1",
+	"params": [
+		{
+			"blockStateCalls": [
+				{
+					"blockOverrides": {
+						"number": "0xb"
+					},
+					"calls": [
+						{
+							"from": "0xc000000000000000000000000000000000000000",
+							"input": "0x4360005260206000f3"
+						}
+					]
+				},
+				{
+					"blockOverrides": {
+						"number": "0xc"
+					},
+					"calls": [
+						{
+							"from": "0xc100000000000000000000000000000000000000",
+							"input": "0x4360005260206000f3"
+						}
+					]
+				}
+			]
+		},
+		"0xa"
+	]
+}
+```
+
+Here we want our calls to be executed in blocks 11 (0xb) and in 12 (0xc). The block numbers can be anything as long as they are increasing and higher than the block we are building from 10 (0xa). So, we could set the blocks to be 100 and 200 for example. Now we end up in a situation where there exists block ranges 13-99 and 101-199 that are not defined anywhere. These blocks are called "phantom blocks". What happens if you try to request block hash of any of such blocks in the EVM? How can we calculate the block hash of future blocks when we don't know the block hash of the previous block?
```

Comment:
I recommend just having block numbers 100 and 200 in the JSON example rather than having sequential numbers and then using text to describe how you might have something different.

Reply:
yeah, That simplifies this. I adjusted it